src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -54,7 +54,8 @@
  (fn [{:keys [db]} [proposal]]
    (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
    (let [accounts                     (get-in db [:wallet :accounts])
-         without-watched              (remove :watch-only? (vals accounts))
+         current-viewing-address      (get-in db [:wallet :current-viewing-account-address])
+         available-accounts           (remove :watch-only? (vals accounts))
```

Comment:
We need to filter the operable accounts (the `operable?` boolean key in the account would be helpful) in addition to removing the watched accounts. The user won't be able to sign any request if the key pair of that account is missing/not imported. I checked the `personal_sign` method from a non-operable account and it failed.

Additionally, we need to remove it from the account selection in the connection request sheet.